From 0b0c5596034a007b71785a0f36aaf40cf74b234d Mon Sep 17 00:00:00 2001
From: Volker Lendecke <vl@samba.org>
Date: Thu, 16 Jun 2011 22:20:49 +0200
Subject: [PATCH] s3: Fix bug 8238 -- KB2536276 prevents access to shares

Without this we were not sending the workgroup name in the negprot reply if
plain text passwords are used.
---
 source3/smbd/negprot.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/source3/smbd/negprot.c b/source3/smbd/negprot.c
index 81d29d9..51ad80d 100644
--- a/source3/smbd/negprot.c
+++ b/source3/smbd/negprot.c
@@ -382,6 +382,7 @@ static void reply_nt1(struct smb_request *req, uint16 choice)
 			reply_nterror(req, NT_STATUS_NO_MEMORY);
 			return;
 		}
+		p += ret;
 		DEBUG(3,("not using SPNEGO\n"));
 	} else {
 		DATA_BLOB spnego_blob = negprot_spnego();
-- 
1.5.6.5

